File: laptop-store-backend/repository/src/main/java/org/example/repository/UserRepository.java
package org.example.repository;

import org.example.model.entity.user.User;
import org.example.model.type.UserRole;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;

public interface UserRepository extends JpaRepository<User, Long> {
    User findByUsername(String username);

    @Query("SELECT u.role FROM User u WHERE u.username = :username")
    UserRole findRoleByUsername(String username);
}

File: laptop-store-backend/repository/src/main/kotlin/org/example/repository/UserRepository.kt
package org.example.repository

import org.example.model.User
import org.example.type.UserRole
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.stereotype.Repository

@Repository
interface UserRepository : JpaRepository<User, Long> {
    fun findByUsername(username: String): User?

    @Query("SELECT u.role FROM User u WHERE u.username = :username")
    fun findRoleByUsername(username: String): UserRole
}